proguard-rules.pro

```diff
@@ -0,0 +1,315 @@
+# Ensure that ProGuard does not strip any fields or methods of classes implementing PersistableProto
+-keepclassmembers class * implements bisq.common.proto.PersistableProto {
+    <fields>;
+    <methods>;
+}
+
+-keep interface bisq.common.proto.Proto { *; }
+-keep class * implements bisq.common.proto.Proto { *; }
+-keep class bisq.persistence.TimestampStore { *; }
+-keep class bisq.persistence.PersistableStoreReaderWriter { *; }
+-keep class bisq.persistence.PersistableStoreReaderWriterTests { *; }
+-keep public class * extends com.google.protobuf.GeneratedMessageLite { *; }
+-keep public class * extends com.google.protobuf.GeneratedMessage { *; }
+-keep class **.protobuf.** { *; }
+-keep class **.protobuf.**$Builder { *; }
+-keepclassmembers class **.protobuf.** { *; }
+-keepclassmembers class **.protobuf.**$Builder { *; }
+
+# Guava
+-dontwarn javax.lang.model.element.Modifier
+-dontnote sun.misc.SharedSecrets
+-keep class sun.misc.** { *; }
+-dontwarn java.lang.SafeVarargs
+-keep class com.google.common.base.Joiner {
+    public static Joiner on(java.lang.String);
+    public ** join(...);
+}
+
+-keep class java.lang.Throwable {
+    void addSuppressed(...);
+}
+
+-keepclassmembers class com.google.common.util.concurrent.AbstractFuture** {
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater waiters;
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater value;
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater listeners;
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater thread;
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater next;
+}
+
+-keepclassmembers class com.google.common.util.concurrent.AtomicDouble {
+    double value;
+}
+
+-keepclassmembers class com.google.common.util.concurrent.AggregateFutureState {
+    int remaining;
+    java.util.concurrent.atomic.AtomicReferenceFieldUpdater seenExceptions;
+}
+
+-dontwarn java.lang.ClassValue
+-dontnote com.google.appengine.api.ThreadManager
+-keep class com.google.appengine.api.ThreadManager {
+    static java.util.concurrent.ThreadFactory currentRequestThreadFactory(...);
+}
+
+-dontnote com.google.apphosting.api.ApiProxy
+-keep class com.google.apphosting.api.ApiProxy {
+    static com.google.apphosting.api.ApiProxy.Environment getCurrentEnvironment(...);
+}
+
+# OkHttp
+-dontwarn okhttp3.**
+-keep class okhttp3.** { *; }
+-dontwarn okio.**
+-keep class okio.** { *; }
+
+#Spring
+-keepclassmembers class * {
+    @org.springframework.beans.factory.annotation.Autowired *;
+}
+-keepclassmembers class * {
+    @javax.annotation.Resource *;
+}
+-keep class javax.annotation.**
+-dontwarn javax.annotation.**
+
+-keepattributes *Annotation*, Signature, EnclosingMethod
+-keep class * extends java.lang.annotation.Annotation { *; }
+-keep class * implements java.lang.reflect.InvocationHandler { *; }
+
+-keep class org.springframework.beans.factory.annotation.**
+-dontwarn org.springframework.beans.factory.annotation.**
```

Comment:
In our code base there is no springframework dependency. I guess its from sub depenendencies. Do you know from where?

Reply:
I added this as part of the "std java project rules" first rule generation, but I guess we can safely remove it. I'll do a quick test and clean it up